packages/core/strapi/lib/core/domain/module/index.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const { validateModule } = require('./validation');
+
+const createModule = (namespace, rawModule, strapi) => {
+  try {
+    validateModule(rawModule);
+  } catch (e) {
+    throw new Error(`strapi-server.js is invalid for plugin ${namespace}.\n${e.errors.join('\n')}`);
+  }
+
+  const loaded = [];
```

Comment:
We need to review this eventually so the logic for loaded is more robust here.

Reply:
we can use an object instead

Reply:
That's ok for now but the issue is we don't garantee that you can't call bootstrap if registered has been called for example :) that's would require a bit more state maangement to implement and it doesn't seem too critical though